### ISSUE TITLE
Bug with xt::sort()?

### ISSUE THREAD

**Issue by DavisVaughan** [QuantStack/xtensor-r#88]
I'm getting some odd results with `xt::sort()` (at least to me they seem odd).

For vectors it seems okay.
For matrices where one of the dimensions is 1, nothing seems to happen.
For matrices where both dims are >1, it seems to sort rows when I would think it would be sorting cols, and vice versa.

All of these are demonstrated below. Am I missing something?

```cpp
// [[Rcpp::depends(xtensorrr)]]
// [[Rcpp::plugins(cpp14)]]

#include <xtensor-r/rarray.hpp>
#include <xtensor/xsort.hpp>
#include <Rcpp.h>
using namespace Rcpp;

// [[Rcpp::export]]
SEXP rray_sort_cpp(xt::rarray<double> x, SEXP arg) {
  std::ptrdiff_t axis = *INTEGER(arg);
  xt::rarray<double> res = xt::sort(x, axis);
  return res;
}
```

``` r
Rcpp::sourceCpp("~/Desktop/test.cpp")

x_vec <- c(1, 3, 4, 2)

x_mat <- matrix(x_vec)
x_mat
#>      [,1]
#> [1,]    1
#> [2,]    3
#> [3,]    4
#> [4,]    2

x_mat2 <- matrix(x_vec, ncol = 2)
x_mat2
#>      [,1] [,2]
#> [1,]    1    4
#> [2,]    3    2

x_mat3 <- t(x_mat)
x_mat3
#>      [,1] [,2] [,3] [,4]
#> [1,]    1    3    4    2

# This looks okay
rray_sort_cpp(x_vec, 0L)
#> [1] 1 2 3 4

# Nothing?
rray_sort_cpp(x_mat, 0L)
#>      [,1]
#> [1,]    1
#> [2,]    3
#> [3,]    4
#> [4,]    2

rray_sort_cpp(x_mat, 1L)
#>      [,1]
#> [1,]    1
#> [2,]    3
#> [3,]    4
#> [4,]    2

# Isn't this supposed to sort cols?
# (it sorts rows)
rray_sort_cpp(x_mat2, 1L)
#>      [,1] [,2]
#> [1,]    1    4
#> [2,]    2    3

# Isn't this supposed to sort rows?
# (it sorts cols)
rray_sort_cpp(x_mat2, 0L)
#>      [,1] [,2]
#> [1,]    1    2
#> [2,]    3    4

# Nothing?
rray_sort_cpp(x_mat3, 0L)
#>      [,1] [,2] [,3] [,4]
#> [1,]    1    3    4    2

rray_sort_cpp(x_mat3, 1L)
#>      [,1] [,2] [,3] [,4]
#> [1,]    1    3    4    2
```

<sup>Created on 2019-01-09 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

**Comment by DavisVaughan** [comment]
> For matrices where both dims are >1, it seems to sort rows when I would think it would be sorting cols, and vice versa.

Actually, I think it is doing this one right. I needed to think about it as "sorting _along_ the axis"

The matrix issue where one of the dimensions is 1 is still a problem, I think.

**Comment by DavisVaughan** [comment]
Intuition is pushing me to think that this is a `column_major` thing, and happens somewhere around here:
https://github.com/QuantStack/xtensor/blob/master/include/xtensor/xsort.hpp#L43

But I don't know the exact issue.

**Comment by wolfv** [comment]
Hi Davis, 

thanks for all your work! thanks to you col-major in xtensor will soon be a first class citizen! 
I think your intuition is correct on this one.

I currently don't have much time to work on xtensor, I'll be more responsive again in February. But if no one else looks into this, I'll try to do it on the weekend.

**Comment by DavisVaughan** [comment]
Fixed by QuantStack/xtensor#1519
